transcom/mymove
# INT-B-21577 Limit UB weight allowance based on Student Travel order type.

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21577)

## Summary

This branch limits the "Unaccompanied Baggage Weight" limit for a move to 350 lbs if the move order is of the "Student Travel" type. Note that currently there is no validation to ensure that Student Travel is only eligible for OCONUS moves, but since the intention is that this will only be an option for OCONUS, this weight limit functionality only works if you choose an origin or destination duty location that is OCONUS (e.g. Alaska).

### How to test

1. Enable the "FEATURE_FLAG_UNACCOMPANIED_BAGGAGE" FF in the .envrc and re-run `make server_run`
2. Create a move with an OCONUS origin/destination duty location (I used Tok, AK) 
3. Create a UB Unaccompanied baggage shipment. (This weight limiting will work with any shipment type since it's on the order level, but the UB shipment gives us an extra UI spot to verify that the limiting is working).
4. Sign for the move, then login as SC and access the move, you should see on the "Allowances" card that the "Unaccompanied Baggage" field shows 350 lbs:![Screenshot 2024-11-26 at 3 07 48 PM](https://github.com/user-attachments/assets/34e357fa-34c5-4474-ba3c-5aae1464583f)
5. Also check the shipment by clicking "Edit Shipment" on the UB shipment card, then verify that this also shows 350 lbs:
![Screenshot 2024-11-26 at 3 10 08 PM](https://github.com/user-attachments/assets/13db7ed9-17b9-4156-a927-a047c1cc4b01)

P.S. you can also create moves without UB and instead have any other shipment type, and you should still see the "Allowances" card show 350 lbs as shown in step 4, you just won't get the UB shipment card as shown in step 5.
